File: dart/06-io-operations/io_files.dart
import 'dart:io';

void main() {
  final file = File('notes.txt');

  // Write text — creates the file, or overwrites it if it already exists
  file.writeAsStringSync('First line\nSecond line\n');

  // Append more text without erasing what's there
  file.writeAsStringSync('Third line\n', mode: FileMode.append);

  // Read the whole file back as one string
  String contents = file.readAsStringSync();
  print('--- File contents ---');
  stdout.write(contents);

  // Read the file as a list of lines
  List<String> lines = file.readAsLinesSync();
  print('The file has ${lines.length} lines.');

  // Clean up after ourselves
  file.deleteSync();
  print('File deleted: ${!file.existsSync()}');
}

File: nim/06-io-operations/io_files.nim
let filename = "notes.txt"

# Write a whole string at once (creates or overwrites the file)
writeFile(filename, "First line\nSecond line\n")

# Append with an explicit file handle opened in append mode
let f = open(filename, fmAppend)
f.writeLine("Third line (appended)")
f.close()

# Read the entire file back as a single string
let contents = readFile(filename)
echo "--- Full contents ---"
echo contents

# Read the file one line at a time (newlines are stripped)
echo "--- Line by line ---"
for line in lines(filename):
  echo "> ", line
